### PR TITLE
Feature #5 clone a product

### PR DESCRIPTION
**Pull request by justindongqiang**
Clone button is on the page of edit a product. 
Click on the button will open the windows has a text box with product name.
User has a opportunity to either cancel it or change the product name and save the product.
This window is important because you don't something like, if user clicks on clone product accidentally, then they don't have a chance to cancel it out.   

image

image

Please have a look at and let me know if you have any questions.

### CODE REVIEW

**Comment by justindongqiang** [comment]
@thiennn any updates?

File: src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/Controllers/ProductCloneApiController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    newProduct.Name = productNmae;
+                    newProduct.CreatedById = currentUser.Id;
+                    newProduct.LatestUpdatedById = currentUser.Id;
+                    newProduct.CreatedOn = DateTimeOffset.Now;
+                    newProduct.LatestUpdatedOn = DateTimeOffset.Now;
+                    _productRepository.Add(newProduct);
+                    _productRepository.SaveChanges();
+                }
+
+                return Ok();
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Catalog/Areas/Catalog/Controllers/ProductCloneApiController.cs]: Generally, we should not return Ok, with empty content. This action should return 201 Created and the url of the product

File: src/Modules/SimplCommerce.Module.Catalog/Models/Product.cs [hunk 2]
```diff
@@ -155,6 +168,33 @@ public Product Clone()
...
 
+            foreach (var media in Medias)
+            {
+                product.AddMedia(new ProductMedia
+                {
+                    MediaId = media.MediaId
+                });
+            }
+
+            foreach (var option in OptionValues)
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Catalog/Models/Product.cs]: I'm not sure whether we should clone the product options and product variation (product link with ProductLinkType.Super).

This behavior can also have side effect when create a new product with variations. This clone method is called for each variation

Comment (justindongqiang) [reply src/Modules/SimplCommerce.Module.Catalog/Models/Product.cs]: okay let me remove the product options and variations clone.

**Comment by justindongqiang** [comment]
@thiennn I have resolved review and please check it and let me know if you have any questions.

**Pull request merged**